dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/WorkflowDefinitionMapper.xml

```diff
@@ -200,10 +200,14 @@
             project_code as projectCode
             ,count(*) as count
         from t_ds_workflow_definition
-        where project_code in
-        <foreach collection="projectCodes" index="index" item="i" open="(" separator="," close=")">
-            #{i}
-        </foreach>
+        <where>
```

Comment:
We should not do this query while projectCodes is null

Reply:
> We should not do this query while projectCodes is null

Is it not possible to query all project when projectCodes is null？？

Reply:
This is used to query the number of workflow definitions under each project. If the projects don't exist, the number of workflow definitions under the projects is meaningless.

Reply:
Yes, we should filter this in upper layer. The project_code should avoid to be null.

Reply:
> Yes, we should filter this in upper layer. The project_code should avoid to be null.

Yes, but this method must also ensure robustness.

Reply:
We don't need this since if the the projects don't exist, the number of workflow definitions under the projects is meaningless.

Reply:
> We don't need this since if the the projects don't exist, the number of workflow definitions under the projects is meaningless.

OK, i have fix it use choose-when-otherwise, maybe this is better.

Reply:
> I think we don't need to change this since we've already filtered it at the entrance.

However, if someone uses this method in the future, there will be error.